todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/source/local/TasksLocalDataSource.kt

```diff
@@ -56,19 +56,20 @@ class TasksLocalDataSource private constructor(context: Context) : TasksDataSour
                 }
                 tasks.add(task)
             }
-            if (tasks.isNotEmpty()) {
-                callback.onTasksLoaded(tasks)
-            } else {
-                // This will be called if the table is new or just empty.
-                callback.onDataNotAvailable()
-            }
             close()
         }
         db.close()
+
+        if (!tasks.isEmpty()) {
```

Comment:
As callback may return at arbitrary time, I thought its better to clean up resources before invoking.